src/D2L.Bmx/BmxConfigProvider.cs

```diff
@@ -48,28 +49,36 @@ internal class BmxConfigProvider : IBmxConfigProvider {
 		);
 	}
 
-	public void SaveConfiguration( BmxConfig config ) {
+	public void SaveConfiguration( BmxConfig config, FileIniDataParser parser ) {
```

Comment:
injecting `FileIniDataParser` via the constructor rather than a method parameter would be more consistent with our general style.

Reply:
I was debating on this and what we'd prefer but thanks I'll swap it it to be from constructor

Reply:
it'd be especially handy once we replace `Microsoft.Extensions.Configuration.Ini` with the ini-parser entirely, in the `GetConfiguration` method too.
https://trello.com/c/bwS24Mo4/1238-replace-microsoftextensionsconfigurationini-with-ini-parser-netstandard